### DISCUSSION TITLE
Returning an error in a middleware when interrupting?

### DISCUSSION THREAD

**Discussion by Atlinx** [twinlogix/typetta#230] [Q&A] [answered]
How can we return an error in middleware? I'm using middleware for validation and I'd like to return an error message whenever validation fails and makes the operation exit early. I'm currently just throwing an error when that happens, but I'm not sure if that's the proper way to do it.

**Comment by edobrb** [comment, marked as answer]
Throwing an exception is the correct way if you have an error. The early interruption of a middleware is to make everything go well without actually making the call on the db. Suppose we want to make a caching middleware, this can work by returning the result of the find in the before function.
